src/hotspot/share/memory/virtualspace.cpp

```diff
@@ -604,16 +610,17 @@ void ReservedHeapSpace::initialize_compressed_heap(const size_t size, size_t ali
     // Last, desperate try without any placement.
     if (_base == nullptr) {
       log_trace(gc, heap, coops)("Trying to allocate at address null heap of size " SIZE_FORMAT_X, size + noaccess_prefix);
-      initialize(size + noaccess_prefix, alignment, page_size, nullptr, false);
+      initialize(size + noaccess_prefix, alignment, page_size, nullptr, !ExecMem, nmt_flag());
     }
   }
 }
 
 ReservedHeapSpace::ReservedHeapSpace(size_t size, size_t alignment, size_t page_size, const char* heap_allocation_directory) : ReservedSpace() {
-
   if (size == 0) {
     return;
   }
+  // _nmt_flag is used internally by initialize_compressed_heap
+  _nmt_flag = mtJavaHeap;
```

Comment:
Nit, we use a mixture of directly accessing _nmt_flag and accessing it via getter. Hotspot seems to prefer getters/setters. Can we use setters here?

Reply:
The flag is not set/changed in other classes, so there is no need to have a `public set_nmt_flag()` member for it.
All the changes to the flag can be done internally using the member directly.
P.S.: There was already a setter but removed after a review comment.